File: src/test/java/warmke/mike/leet/p7/SolutionTest.java
package warmke.mike.leet.p7;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import warmke.mike.leet.bit.p7.Solution;

import static org.junit.jupiter.api.Assertions.assertEquals;


public class SolutionTest {

    Solution solution;

    @BeforeEach
    public void setup() {
        solution = new Solution();
    }

    @Test
    public void test() {
        assertEquals(321, solution.reverse(123));
        assertEquals(-321, solution.reverse(-123));
        assertEquals(21, solution.reverse(120));
        assertEquals(0, solution.reverse(0));
        var minVal = -(int) Math.pow(2, 31);
        assertEquals(0, solution.reverse(minVal));
        var maxVal = ((int) Math.pow(2, 31)) - 1;
        assertEquals(0, solution.reverse(Integer.MIN_VALUE));
    }
}

File: src/test/java/warmke/mike/leet/p238/SolutionTest.kt
package warmke.mike.leet.p238

import org.junit.jupiter.api.Test
import warmke.mike.leet.arrays.p238.Solution
import kotlin.test.assertEquals

class SolutionTest {

    val s = Solution()

    @Test
    fun test() {
        assertEquals(intArrayOf(24,12,8,6), s.productExceptSelf(intArrayOf(1,2,3,4)))
    }
}